ocaml/lambda/translcore.ml

```diff
@@ -145,8 +145,8 @@ let may_allocate_in_region lam =
   let rec loop_region lam =
     shallow_iter ~tail:(function
       | Lexclave body -> loop body
-      | _ -> ()
-    ) ~non_tail:(fun _ -> ()) lam
+      | lam -> loop_region lam
+    ) ~non_tail:(fun lam -> loop_region lam) lam
```

Comment:
Unnecessary eta-expansion

Reply:
I'm inclined to keep it expanded - the code is a bit clearer when the two recursive cases are symmetric